batch/batch/aioclient.py

```diff
@@ -152,36 +287,31 @@ def __init__(self, client, id, attributes):
         if pvc_size:
             doc['pvc_size'] = pvc_size
 
-        j = await self._client._post('/jobs/create', json=doc)
+        self._job_docs.append(doc)
 
-        return Job(self,
-                   j['job_id'],
-                   attributes=j.get('attributes'),
-                   parent_ids=j.get('parent_ids', []))
+        j = Job.unsubmitted_job(self, self._job_idx, attributes, parent_ids)
+        self._jobs.append(j)
+        return j
 
-    async def close(self):
-        await self._client._patch(f'/batches/{self.id}/close')
+    async def submit(self):
+        if self._submitted:
+            raise ValueError("cannot submit an already submitted batch")
+        self._submitted = True
 
-    async def cancel(self):
-        await self._client._patch(f'/batches/{self.id}/cancel')
+        if len(self._job_docs) != 0:
+            self._doc['jobs'] = self._job_docs
 
-    async def status(self):
-        return await self._client._get(f'/batches/{self.id}')
+        b = await self._client._post('/batches/create', json=self._doc)
+        batch = Batch(self._client, b['id'], b.get('attributes'))
 
-    async def wait(self):
-        i = 0
-        while True:
-            status = await self.status()
-            if status['complete']:
-                return status
-            j = random.randrange(math.floor(1.1 ** i))
-            time.sleep(0.100 * j)
-            # max 4.45s
-            if i < 64:
-                i = i + 1
+        for j in self._jobs:
+            j._job = j._job._submit(batch)
```

Comment:
It seems somewhat cleaner to just 

```
# Nothing created
for doc in self._job_docs:
   # Adding jobs
   self._jobs.append(Jobs.create_job(batch, doc))
```

Reply:
And I think you're planning on doing the insertion in a batched way (either all insertions happening in one query, or some chunks from the full range for very large jobs, to avoid running up against some request size limit)